DB/GADPerson.m

```diff
@@ -89,14 +92,49 @@ -(GADPerson *)initWithDictionary: (NSDictionary *) dict {
     person3.office_email=@"sgaprez";
     person3.office_hours=@[@"Sunday: 7-8pm Grille, 8-9pm JRC 222",@"  Monday: 9 - 3 Commons",@"  Tuesday: 7 -7 JRC Main",@"  Wednesday: 12 - 12 Raynard's Office"];
 
-    
     people=@[person1,person2,person3];
     
     return people;
 }
 
++ (void) fetchPersonInfoWithCriteria:(NSDictionary*)criteria Username:(NSString*)usrename Password:(NSString*) password completionHandler:(void(^_Nonnull)(NSArray<GADPerson *> *))completion {
+
+    NSMutableArray *queryItems = [NSMutableArray<NSURLQueryItem *> new];
+    for (NSString *key in criteria){
+        [queryItems addObject:[NSURLQueryItem queryItemWithName:key value:criteria[key]]];
+    }
+    NSURLComponents *components = [NSURLComponents componentsWithString:apiURL];
+    components.queryItems=queryItems;
+    NSURL *url = components.URL;
+    
+    NSString *authenticationString = [NSString stringWithFormat:@"{'un':'%@', 'pw':'%@'}",usrename,password];
+    NSData *bodyData = [authenticationString dataUsingEncoding:NSUTF8StringEncoding];
+    NSString *bodyLength = [NSString stringWithFormat:@"%lu",(unsigned long)[bodyData length]];
+    
+    NSMutableURLRequest *req = [NSMutableURLRequest requestWithURL:url cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:timeoutInterval];
+    [req setHTTPMethod:@"POST"];
+    [req setValue:bodyLength forHTTPHeaderField:@"Content-Length"];
+    [req setValue:@"application/json" forHTTPHeaderField: @"Content-Type"];
+    [req setHTTPBody:bodyData];
+    
+    NSURLSessionDataTask *task = [[NSURLSession sharedSession]dataTaskWithRequest:req completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (error) {
+            NSLog(@"EEERRRRORRRRR");
+            return;
+        }
+        NSMutableArray <GADPerson*> *people=[NSMutableArray<GADPerson*> new];
+        NSError *JSONParsingError = [[NSError alloc]initWithDomain:NSCocoaErrorDomain code: NSPropertyListReadCorruptError userInfo:nil];
```

Comment:
You don't actually need to set this to anything. Just `NSError *JSONParsingError;` here and the rest should work as expected.